Preethi

ppreethi89 opened the pull request:
with levels

borisv13 commented on src/main/java/movement/Movement.java:
```diff
@@ -0,0 +1,43 @@
+
+	private int level = 0;
+
+	public int setLocation(Point player, Map<Point, TileType> tiles, int x, int y, int level) {
```
1. Shouldn't have to pass in Player and Map - easier to just pass in GameEngine and retrieve the data from it directly.
2. The names x and y are bad names, do not indicate what their significance is.
3. Why is int level passed in to a method called setLocation?
4. This method violates SRS by also manipulating level.

borisv13 commented on src/main/java/engine/GameEngine.java:
```diff
@@ -74,19 +76,41 @@ public int getPlayerYCoordinate() {
+
+	public void setLocation(int x, int y) {
+
+		Movement movement = new Movement();
```
I can tell this was not Test Driven by the fact that this line exists.
If an object is created inline like this, it demonstrates a violation of Dependency Inversion principle - ie there is no way to mock Movement now because it's not being handed off.
The correct way to do this is to create Movement all the way at the top (ObjectFactory) and pass it in to GameEngine when that is being created.

ppreethi89 commented:
I am little confused while making these changes. I have added levelHandler to provide increment level count and to check if its last level. And LevelCreator will just create level based on this. But instead of increasing level count in levelHandler if I do it in gameEngine itself, will that be against SRP? As anyways its the game engines property, will it be wrong to increment there?

borisv13 commented on src/main/java/parser/LevelCreator.java:
```diff
@@ -17,20 +17,34 @@
-			reader = readerWrapper.createBufferedReader(getFilePath(level));
+			int newLevel = levelHandler.getNewLevel(level);
+
+			if (!levelHandler.checkIfLastLevel(newLevel)) {
```
Is it possible to extract this if/else into a separate method?

borisv13 commented on src/main/java/engine/GameEngine.java:
```diff
@@ -74,19 +77,35 @@ public int getPlayerYCoordinate() {
 	public void keyLeft() {
-		// TODO Implement movement logic here
+
+		setLocation((int) player.getX() - 1, (int) player.getY());
```
This seems outdated per master. Please see master keyLeft either use that or explain why this is different.

borisv13 replied:
This is still outstanding.

ppreethi89 replied:
I am still working on this, the move left of my code is slightly different from master, as in my features, I am testing the levels and positions as absolute value.. for ex, to check if a player reaches 3,3 which is Win_point, then his level should next_level. If i use master code, i need to change my when condition in features, to traverse to that position.i

I am working to change the movement code and corresponding feature file for this

ppreethi89 replied:
this is fixed, gameengine is using master code

ppreethi89 commented:
I have moved the loop into different function. Create custom exception for win message, so in later stage can use that message for display.
the move left of my code is slightly different from master, as in my features, I am testing the levels and potions as absolute value.. for ex, to check if a player reaches 3,3 which has Win_point, if I use master code, then I have to move the player many times to go to that coordinates, which is why I have modified it this way. If this is not good, I have to change my test approach to suit master file method

ppreethi89 commented:
I am thinking to add some points to player that he can collect while he travels to this level and after last level gets completed, it should display winning message instead of exiting. Shall I proceed to add this feature to my code

borisv13 commented on src/main/java/tiles/TileType.java:
```diff
@@ -2,7 +2,7 @@
 public enum TileType {
 
-	PASSABLE(' '), NOT_PASSABLE('X'), PLAYER('P');
+	PASSABLE(' '), NOT_PASSABLE('X'), PLAYER('P'), WIN_POINT('E'), GAMEOVER('G');
```
I don't see GAMEOVER being tested or used anywhere.

ppreethi89 replied:
I added Gameover, out of confusion of good name for exit point of level. retained win_point and missed to remove this, this is removed now.

ppreethi89 commented:
That's my mistake, i was confused on right name for my End point of level, and forgot to remove the older one 'GAMEOVER' after deciding to go with 'WIN_POINT'

ppreethi89 commented:
i removed unused code section, which was autogenerated while changing access modifier. Checked all coverage is now 100%.
